db/columns/operations/alter.py

```diff
@@ -103,12 +103,22 @@ def retype_column(
     except DataError as e:
         if type(e.orig) == InvalidParameterValue:
             raise InvalidTypeOptionError
-        if type(e.orig) == InvalidTextRepresentation:
-            raise InvalidTypeError
+        elif type(e.orig) == InvalidTextRepresentation:
+            raise InvalidTypeError(column_name, new_type)
+        elif type(e.orig) == StringDataRightTruncation:
+            raise InvalidStringTruncation
         else:
-            raise e
+            raise e.orig
     except InternalError as e:
-        raise e.orig
+        if type(e.orig) == RaiseException:
+            raise InvalidTypeError(column_name, new_type)
+        else:
+            raise e.orig
+    except ProgrammingError as e:
+        if type(e.orig) == SyntaxError:
+            raise InvalidTypeOptionError
+        else:
+            raise e.orig
```

Comment:
I'm suspicious of this.
- How is this being produced?
- Is `SyntaxError` really precise enough?

Reply:
`SyntaxError` occurs when we pass something off limits like 99999999 in the Field size limit.

Reply:
My concern is whether some other thing can also cause a `SyntaxError`. After looking at the docs a bit (and what's reported by `psycopg2`), it doesn't seem like it's possible to do much better.